packages/broker/src/plugins/operator/MaintainOperatorValueService.ts

```diff
@@ -1,81 +1,49 @@
 import { Logger, scheduleAtInterval } from '@streamr/utils'
 import { MaintainOperatorValueHelper } from './MaintainOperatorValueHelper'
 import { OperatorServiceConfig } from './OperatorPlugin'
-import { BigNumber } from 'ethers'
 
 const logger = new Logger(module)
 
-const CHECK_VALUE_INTERVAL = 1000 * 60 * 60 * 24 // 1 day
-const ONE_ETHER = BigInt(1e18)
+const DEFAULT_CHECK_VALUE_INTERVAL_MS = 1000 * 60 * 60 * 24 // 1 day
 
 export class MaintainOperatorValueService {
-    private penaltyLimitFraction: bigint
+    private readonly withdrawLimitSafetyFraction: bigint
+    private penaltyLimitFraction?: bigint
     private readonly helper: MaintainOperatorValueHelper
     private readonly abortController: AbortController
+    private readonly checkIntervalMs: number
 
-    constructor(config: OperatorServiceConfig, penaltyLimitFraction = BigInt(0)) {
-        this.penaltyLimitFraction = penaltyLimitFraction
+    constructor(config: OperatorServiceConfig, withdrawLimitSafetyFraction = 0.5, checkValueIntervalMs = DEFAULT_CHECK_VALUE_INTERVAL_MS) {
+        this.withdrawLimitSafetyFraction = BigInt(withdrawLimitSafetyFraction * 1e18)
         this.helper = new MaintainOperatorValueHelper(config)
         this.abortController = new AbortController()
+        this.checkIntervalMs = checkValueIntervalMs
+    }
+
+    async checkMyUnwithdrawnEarnings(): Promise<void> {
+        logger.info('Check unwithdrawn earnings % of total value, and check if it\'s above the safe threshold')
```

Comment:
This is good to log imo (since it happens only hourly), can we make the log message a bit more succinct so it is clear to the end-user what is happening?

Maybe something like?
```suggestion
        logger.info('Check whether it is time to withdraw my earnings')
```

Reply:
Changed